src/fvm_multicell.hpp

```diff
@@ -184,7 +178,32 @@ class fvm_multicell {
 
     std::size_t num_probes() const { return probes_.size(); }
 
+    //
+    // Threshold crossing interface.
+    // Used by calling code to perform spike detection
+    //
+
+    /// types defined by the back end for threshold detection
+    using threshold_watcher = typename backend::threshold_watcher;
+    using crossing_list     = typename backend::threshold_watcher::crossing_list;
+
+    /// Forward the list of threshold crossings from the back end.
+    /// The list is passed by value, because we don't want the calling code
+    /// to depend on references to internal state of the solver, and because
+    /// for some backends the results might have to be collated before returning.
+    crossing_list get_spikes() const {
```

Comment:
This is a bit awkward, because it is spike-specific code pretending to be generic! I went with calling them "crossings" in the watcher, to make generalizing the watcher simpler in the future.